src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/CreateGeneralLetterAboutToSubmitHandler.java

```diff
@@ -39,7 +39,9 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
         validateCaseData(callbackRequest);
 
         if (generalLetterService.getCaseDataErrorsForCreatingPreviewOrFinalLetter(caseDetails).isEmpty()) {
+            generalLetterService.addFrcCourtFields(caseDetails);
```

Comment:
This approach seems to make the client code more complex than it needs to be. You could hide the adding and removing of FRC court fields behind the API:
`generalLetterService.previewGeneralLetterWithFrcCourtFields`

Are the addFrcCourtFields and removeFrcCourtFields going to be used in other situations other than for the purposes of generating a letter?